contracts/src/BeefyClient.sol

```diff
@@ -455,7 +455,9 @@ contract BeefyClient {
             numRequiredSignatures += 1 + 2 * Math.log2(signatureUsageCount, Math.Rounding.Ceil);
         }
 
-        return Math.min(numRequiredSignatures, validatorSetLen);
+        // Never require more signatures than 2/3 majority.
+        uint256 validatorSetMajority = validatorSetLen - (validatorSetLen - 1) / 3;
+        return Math.min(numRequiredSignatures, validatorSetMajority);
```

Comment:
Are there unit tests that test these conditions?

Reply:
This cases:
https://github.com/Snowfork/snowbridge/pull/956/commits/90923476fb11fe7016dea487b4123d06331bc171#diff-87ab70972cd5c5d114ecf33d954f8a1880f72747fbbebb53db2141bd93203737R686-R688

But I have added a few more to the same test.